### PR TITLE
JAMES-2949 Remove FIXME-USERNAME annotations by using lower case comparison

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[0957358f12b0cde55027df068b0c0d12b091b536] [INFO] There are 2 errors reported by Checkstyle 8.23 with checkstyle.xml ruleset.
[0957358f12b0cde55027df068b0c0d12b091b536] [ERROR] src/main/java/org/apache/james/smtpserver/SenderAuthIdentifyVerificationRcptHook.java:[21,8] (imports) UnusedImports: Unused import - java.text.Collator.
[0957358f12b0cde55027df068b0c0d12b091b536] [ERROR] src/main/java/org/apache/james/smtpserver/SenderAuthIdentifyVerificationRcptHook.java:[22,8] (imports) UnusedImports: Unused import - java.util.Locale.
```

File: core/src/main/java/org/apache/james/core/Username.java [hunk 1]
```diff
@@ -125,6 +125,17 @@ public String asId() {
         return asString().toLowerCase(Locale.US);
     }
 
+    public boolean equalsAsId(String otherId) {
```
Comment (trantienduchn) [inline core/src/main/java/org/apache/james/core/Username.java]: is there any reason you don't override equals() but creating this method?

Comment (chibenwa) [reply core/src/main/java/org/apache/james/core/Username.java]: I can answer you: username is case sensitive, it is its usage within UsersReposiotory which is case insentive, hence we should not alter equals with such concerns (source of potential wrong tests and thus bugs)

Comment (trantienduchn) [reply core/src/main/java/org/apache/james/core/Username.java]: I still don't feel comfortable with more than one equals... method in a POJO, should the `Id` be strong typed?

Comment (chibenwa) [reply core/src/main/java/org/apache/james/core/Username.java]: POJO is the right place to put some domain logic.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**